src/DFTK.jl

```diff
@@ -164,12 +164,24 @@ export kgrid_from_minimal_spacing, kgrid_from_minimal_n_kpoints
 include("symmetry.jl")
 include("bzmesh.jl")
 
+export DensityMethod
+export AtomicDensity
+export RandomDensity
+export CoreDensity
+export ValenceGaussianDensity
+export ValenceNumericalDensity
+export ValenceAutoDensity
 export guess_density
 export random_density
+export atomic_density
+export atomic_total_density
+export atomic_spin_density
+export atomic_density_superposition
```

Comment:
Do we really need all of these functions exported?

It feels a bit duplicate with having both the types and the functions to select behaviour.  I think making the API a bit thinner and try to see if some code can be cut in density_methods.jl might be good (can also happen in a follow up -> put on PseudoIO issue)

Reply:
I think this fits better with the work mentioned in #825 on refactoring form factor / atomic superposition Fourier transform calculations.

I don't think they all need to be exported, probably just guess and random density.